longevity_sla_test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+import re
+
+from longevity_test import LongevityTest
+from sdcm.sct_events import Severity
+from sdcm.sct_events.system import TestFrameworkEvent
+from test_lib.sla import Role, ServiceLevel
+
+
+class LongevitySlaTest(LongevityTest):
+    DEFAULT_USER = "cassandra"
+    DEFAULT_USER_PASSWORD = "cassandra"
+    STRESS_ROLE_NAME_TEMPLATE = 'role%d'
+    STRESS_ROLE_PASSWORD_TEMPLATE = 'rolep%d'
+    SERVICE_LEVEL_NAME_TEMPLATE = 'sl%d'
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.service_level_shares = self.params.get("service_level_shares")
+        self.roles = []
+
+    def test_custom_time(self):
+        with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0], user=self.DEFAULT_USER,
+                                                    password=self.DEFAULT_USER_PASSWORD) as session:
+            for shares in self.service_level_shares:
+                self.roles.append(self.create_sla_auth(session=session, shares=shares))
+
+        self.add_sla_credentials_to_stress_cmds()
+        super().test_custom_time()
+
+    def create_sla_auth(self, session, shares):
+        role = Role(session=session, name=self.STRESS_ROLE_NAME_TEMPLATE % shares,
+                    password=self.STRESS_ROLE_PASSWORD_TEMPLATE % shares, login=True).create()
+        role.attach_service_level(ServiceLevel(session=session, name=self.SERVICE_LEVEL_NAME_TEMPLATE % shares,
```

Comment:
This might fail for cluster greater than 3 nodes since the information might not have been propagated to the node that
does the attachment. 
You must wait here until all nodes have added the service level or it will be flaky.

Reply:
@eliransin how can I check that all nodes have added the service level? 
Is running `LIST ATTACHED SERVICE_LEVEL OF <role_name>;` enough? Do I need to connect to every node and check it on every node?

Reply:
There is a log entry on each node (`service level "XXX" was added.`)

Reply:
Another method could be to retry the attach for  ~25-30 seconds until succeeds. If it does succeed it means that 
everything is fine and the configuration was used.

Reply:
Added retrying